src/main/java/com/zepben/cimbend/get/NetworkConsumerClient.kt

```diff
@@ -96,60 +103,117 @@ class NetworkConsumerClient(
         }
     }
 
+    /***
+     * Retrieve the feeder network for the specified [mRID] and store the results in the [service].
+     *
+     * This is a convenience method that will fetch the feeder object, all of the equipment referenced by the feeder (normal state),
+     * the terminals of all elements, the connectivity between terminals, the locations of all elements, the ends of all transformers
+     * and the wire info for all conductors.
+     *
+     * @return The [Feeder], or null if it was nto found.
+     */
+    fun getFeeder(service: NetworkService, mRID: String): GrpcResult<Feeder> {
+        return tryRpc {
+            val feederResponse = getIdentifiedObject(service, mRID)
+            val feeder = feederResponse.result
+
+            if (!feederResponse.wasSuccessful)
+                throw feederResponse.thrown!!
+            else if (feeder == null)
+                return@tryRpc GrpcResult.of(null)
+            else if (feeder !is Feeder)
+                throw ClassCastException("Unable to extract feeder network from ${feeder.typeNameAndMRID()}.")
+
+            validateRpcCall(getIdentifiedObjects(service, service.getUnresolvedReferenceMrids(Resolvers.equipment(feeder))))
+
+            val mRIDs = service.getUnresolvedReferenceMrids(Resolvers.normalEnergizingSubstation(feeder)).toMutableSet()
+
+            feeder.equipment.forEach {
+                if (it is ConductingEquipment) {
+                    it.terminals.forEach { terminal ->
+                        mRIDs.addAll(service.getUnresolvedReferenceMrids(Resolvers.connectivityNode(terminal)))
+                    }
+                }
+
+                if (it is Conductor) {
+                    if (it is AcLineSegment)
+                        mRIDs.addAll(service.getUnresolvedReferenceMrids(Resolvers.perLengthSequenceImpedance(it)))
+                    mRIDs.addAll(service.getUnresolvedReferenceMrids(Resolvers.assetInfo(it)))
+                }
```

Comment:
This is not an else if block, it will run both parts with a `Conductor`, with an additional line if it is an `AcLineSegment` (only option for now, but maybe not forever)

If you have a particular `when` block you think works, can you please post it

Reply:
Which makes me think do we just need a BaseService.getUnresolvedReferenceMrids(idObj: IdentifiedObject) so you can just get all the unresolved references for a specific object if you want to easily resolve everything?

Reply:
@gcarwardine I am happy for a new conversation not linked the the PR to be created to discuss any changes to the getUnresolvedReferenceMrids interface, it is more than what I want to put in this PR.

Reply:
herp derp of course, no worries